### ISSUE TITLE
user asked twice for a gpg password after a copy-message cmd (shouldn't be asked at all)

### ISSUE THREAD

**Issue by libBletchley** [neomutt/neomutt#898]
* **bug reports**

    * Expected behavior
When on the index page with the cursor placed on an encrypted msg (indicated by a "P"), and the `copy-message` command is given using "C", mutt should copy the message to a file without prompting for a any password because it should not be running gpg.  The `copy-message` surely must differ from the `decrypt-copy` command.

    * Actual behavior
Neomutt prompts for the password.  Then after supplying the password gpg itself prompts for the password graphically.  So the user is needlessly nagged **twice** for a password that is not used.
The file it produces is the original ciphertext, so the gpg output isn't being used.

    * Steps to reproduce
1. Make sure this config setting is in play:
```
set pgp_decrypt_command="/usr/bin/gpg --decrypt --passphrase-fd 0 --pinentry-mode loopback --no-verbose --quiet  --batch  --output - %f"
```
2. Go to the index page
3. Put the cursor on a "P"-indicated message
4. Press "C"

    * Used program versions
    failing version: NeoMutt 20170113 (the current Debian Stretch version)
    working version: whatever version of mutt is distributed with debian jessie

    * Operating System and its version
    Debian Stretch

Note there are two bugs here:

1. The GUI password entry is apparently a new feature of gpg (probably versions > 2.1), and this is causing redundant prompting b/c mutt has its builtin-password prompting.

2. The execution of gpg at all in this scenario is a bug.  While it should be fixed, care should be taken to fix bug 1 (above) first, to make sure bug 1 isn't simply hidden by this fix.

**Comment by libBletchley** [comment]
**workaround** Enabling gpgme by configuring with `set crypt_use_gpgme = yes` will avoid the redundant password entry.  But it's not a solution because gpgme breaks other things, like ability to send encrypted messages to hushmail and protonmail users (gpgme breaks traditional inline pgp).

**Comment by gahr** [comment]
I cannot reproduce. Copy message here correctly copies the encrypted message without trying to decrypt it.

Again, please try with a recent version of neomutt, trim your config to a minimum, and try to incrementally add to it until you figure which setting is causing the misbehaviour. Thanks!

**Comment by libBletchley** [comment]
First of all, I need to be clear that the output is correct.  Mutt produces the ciphertext, as it should.  The problem is the mere prompting for a GPG password.  It asks for a password, and perhaps it uses the password internally, but the GPG output is not used.

BTW, I just tested with a minimal config, both NeoMutt `20170113 (1.7.2)` and NeoMutt `20171013-49-42f391` and the bug does not manifest in either of them.  So it must be an interaction with my config.

**Comment by libBletchley** [comment]
So apparently I need to run `copy-message` many times with different configs.  Is there a way I can automate that?  I don't suppose I can do something like `mutt -e copy-message <msgid> output.mbox`?

**Comment by libBletchley** [comment]
So what I've found is quite baffling.  These two settings togther are all that's needed to reproduce the bug:
```
unset crypt_use_gpgme
save-hook  '~b "Content-Type: text/html;charset=euc-kr"' +SPAM
```
That hook is not my idea, just something I inherited when copying Sven Guckes configs to have a starting point.  If I `grep -i euc-kr my.mbox`, there are no matches, so it's astonishing that a save-hook that would never trigger is causing gpg to execute.
